p2p/src/serv.rs

```diff
@@ -175,13 +174,21 @@ impl Server {
 		let total_diff = self.peers.total_difficulty();
 
 		// accept the peer and add it to the server map
-		let mut peer = Peer::accept(
+		let mut peer = match Peer::accept(
 			&mut stream,
 			self.capabilities,
 			total_diff,
 			&self.handshake,
 			self.peers.clone(),
-		)?;
+		) {
+			Ok(p) => p,
+			Err(e) => {
+				// in theory, should be shutdown when stream gets dropped,
+				// practically doesn't seem to be happening
+				let _ = stream.shutdown(Shutdown::Both);
+				return Err(e);
+			}
+		};
```

Comment:
this seems not necessary, since we already have the error handling in function `accept` and did that `shutdown`.

Reply:
Looks like you're right, good catch! I'm still confused because after a while I end up with some peers having 20+ connections and not showing at all in my peer list. I'll try to track that down some more.